features/issues/api/use-get-issues.tsx

```diff
@@ -2,32 +2,83 @@ import { useEffect } from "react";
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { getIssues } from "@api/issues";
 import type { Page } from "@typings/page.types";
-import type { Issue } from "@api/issues.types";
+import { Issue, IssueLevel, IssueStatus } from "@api/issues.types";
 
 const QUERY_KEY = "issues";
 
-export function getQueryKey(page?: number) {
-  if (page === undefined) {
-    return [QUERY_KEY];
+export function getQueryKey(
+  page?: number,
+  status?: string,
+  level?: string,
+  project?: string,
+) {
+  return [QUERY_KEY, page, status, level, project];
+}
+
+// Function to transform IssueStatus to API status
+function toApiIssueStatus(status: IssueStatus | undefined): string | undefined {
+  switch (status) {
+    case IssueStatus.unresolved:
+      return "open";
+    case IssueStatus.resolved:
+      return "resolved";
+    default:
+      return undefined;
+  }
+}
+
+// Function to transform API status to IssueStatus
+function fromApiIssueStatus(
+  status: string | undefined,
+): IssueStatus | undefined {
+  switch (status) {
+    case "open":
+      return IssueStatus.unresolved;
+    case "resolved":
+      return IssueStatus.resolved;
+    case IssueStatus.unresolved:
+      return status;
+    default:
+      return undefined;
```

Comment:
Similar to a comment in one of the other PRs: I think these are data transformations from the API types to values that are displayed in the UI, right? I would do that at the component level. Then the whole application uses the same domain or core data models (like the Issue type returned from the API). Only close to the UI you'd transform the core fields to display values like "Critical" or here "Open" instead of "unresolved"